### PR TITLE
bpo-42431: Fix outdated bytes comments

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
Also move definitions of internal macros F_LJUST etc to private header.

https://bugs.python.org/issue42431

### CODE REVIEW

File: Include/cpython/bytesobject.h [hunk 1]
```diff
@@ -2,6 +2,9 @@
 #  error "this header file must not be included directly"
 #endif
 
+/* Caching the hash (ob_shash) saves recalculation of a bytes' hash value.
+   This significantly speeds up dict lookups. */
```
Comment (vstinner) [inline Include/cpython/bytesobject.h]: Maybe move this comment before ob_shash definition, in the structure.

File: Include/internal/pycore_format.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#ifndef Py_INTERNAL_FORMAT_H
+#define Py_INTERNAL_FORMAT_H
+#ifdef __cplusplus
+extern "C" {
+#endif
```
Comment (vstinner) [inline Include/internal/pycore_format.h:5]: I'm fine with adding pycore_format.h.

But maybe we could rename pycore_bytes_methods.h to pycore_bytes.h or pycore_bytesobject.h and put everything in a single header file. It's up to you.

It seems like unicode doesn't use these constants, so maybe be more explicit in the file name and rename it to "pycore_bytesformat.h" or "pycore_bytes_format.h"?

File: Include/internal/pycore_format.h [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+#ifndef Py_BUILD_CORE
+#  error "this header requires Py_BUILD_CORE define"
+#endif
+
+/* Format codes
+ * F_LJUST      '-'
+ * F_SIGN       '+'
+ * F_BLANK      ' '
+ * F_ALT        '#'
+ * F_ZERO       '0'
```
Comment (vstinner) [inline Include/internal/pycore_format.h:16]: Would you mind to give an example of function which uses these constants?

File: Objects/bytesobject.c [hunk 4]
```diff
@@ -5,6 +5,7 @@
 #include "Python.h"
 #include "pycore_abstract.h"      // _PyIndex_Check()
 #include "pycore_bytes_methods.h" // _Py_bytes_startswith()
+#include "pycore_format.h"        // F_LJUST etc
```
Comment (vstinner) [inline Objects/bytesobject.c]: Remark: I have a personal preference to only give the name of a single symbol which is "imported" (included). I do that to check sometimes if the include is still needed. One symbol is enough (not "etc.") ;-)

File: Include/cpython/bytesobject.h (hunk 1 above)
Comment (serhiy-storchaka) [reply Include/cpython/bytesobject.h]: Or just remove it as trivial? We do not have similar note for other cached hashes, and initially this comment was about strings.

File: Include/internal/pycore_format.h (hunk 2 above)
Comment (serhiy-storchaka) [reply Include/internal/pycore_format.h:5]: These constants are used both by bytes and unicode formatters. This is the reason why they are declared in the header file.

`pycore_bytes_methods.h` contains declarations for `bytes_methods.c` which contains common code and docstrings for bytes and bytearray.

File: Include/internal/pycore_format.h (hunk 3 above)
Comment (serhiy-storchaka) [reply Include/internal/pycore_format.h:16]: They are used in a number of internal functions. Their names do not matter because they are static.

**Pull request merged**